### PR TITLE
Add setting to control whether to focus the inline editor in peek widget by default

### LINKED ISSUE #23001: Focus peek window on creation
I _love_ the peek window. It's pretty much the best thing ever.

I would love it ever so slightly more if VSCode would auto-focus it and insert the cursor where it belongs. e.g. if I clicked "peek definition", the cursor should be on the definition of what I peeked. 

This idea seems so obvious I can't believe it's not part of the default implementation, but I know you guys have your reasons for doing things. 😄 

If there is such a reason, could this at least be made a setting?

Thanks guys! ❤️

### PR DESCRIPTION
**Pull request by jeanp413**
This PR fixes #23001

### CODE REVIEW

File: src/vs/editor/contrib/gotoSymbol/peek/referencesController.ts [hunk 1]
```diff
@@ -218,7 +225,7 @@ export abstract class ReferencesController implements editorCommon.IEditorContri
 		this._requestIdPool += 1; // Cancel pending requests
 	}
 
-	private _gotoReference(ref: Location): Promise<any> {
+	private _gotoReference(ref: Location): Promise<void> {
```
Comment (jrieken) [inline src/vs/editor/contrib/gotoSymbol/peek/referencesController.ts]: unrelated change? (this and the two below)

**Review by jrieken — CHANGES_REQUESTED** [review]

File: src/vs/editor/contrib/gotoSymbol/peek/referencesController.ts [hunk 2]
```diff
@@ -200,10 +204,13 @@ export abstract class ReferencesController implements editorCommon.IEditorContri
 		}
 		const target = this._model.nextOrPreviousReference(source, fwd);
 		const editorFocus = this._editor.hasTextFocus();
+		const previewEditorFocus = this._widget.isPreviewEditorFocused();
 		await this._widget.setSelection(target);
 		await this._gotoReference(target);
 		if (editorFocus) {
 			this._editor.focus();
+		} else if (this._widget && previewEditorFocus) {
+			this._widget.focusOnPreviewEditor();
```
[the remark this answers is not in the document]
Comment (jeanp413) [reply src/vs/editor/contrib/gotoSymbol/peek/referencesController.ts]: `this._widget` could be undefined if inside `this._gotoReference` it goes through here ⬇️ , though not sure how to trigger that from within the editor.
https://github.com/microsoft/vscode/blob/a78e5830fd58bead0d190e689e16097bbdb669c9/src/vs/editor/contrib/gotoSymbol/peek/referencesController.ts#L252-L257

[the remark this answers is not in the document]
Comment (jrieken) [reply src/vs/editor/contrib/gotoSymbol/peek/referencesController.ts]: Check

File: src/vs/editor/common/config/commonEditorConfig.ts [hunk 3]
```diff
@@ -489,6 +489,11 @@ const editorConfiguration: IConfigurationNode = {
 			default: false,
 			markdownDescription: nls.localize('stablePeek', "Keep peek editors open even when double clicking their content or when hitting `Escape`.")
 		},
+		'editor.peekWidgetFocusInlineEditor': {
+			type: 'boolean',
+			default: false,
+			markdownDescription: nls.localize('peekWidgetFocusInlineEditor', "Controls whether to focus the inline editor in the peek widget by default.")
+		},
```
[the remark this answers is not in the document]
Comment (jrieken) [reply src/vs/editor/common/config/commonEditorConfig.ts]: Sorry, the stablePeek setting wasn't actually a good sample. Try [`suggestFontSize`](https://github.com/microsoft/vscode/blob/8349ba385a89c3cf62c51de405a849e06a588fbc/src/vs/editor/common/config/editorOptions.ts#L3603). You need to add the property to `IEditorOptions`, `EditorOptions` and `EditorOption`

[the remark this answers is not in the document]
Comment (jeanp413) [reply src/vs/editor/common/config/commonEditorConfig.ts]: Done 👍

**Review by jrieken — APPROVED** [review]
Looking good. Not super happy with the name of the setting yet, but we can tweak that at a later time - unless you have spontaneous idea.

**Pull request merged**